src/transactions/association.ts

```diff
@@ -36,11 +39,35 @@ export function invokeAssociation(paramsOrTx: any, seed?: TSeedTypes): IAssociat
     hash: paramsOrTx.hash ? paramsOrTx.hash : '',
   }
 
+  const v3Tx: IAssociationTransactionV3 & WithId & WithProofs = {
+    id: '',
+    type,
+    version,
+    chainId,
+    sender,
+    senderPublicKey,
+    // Not used as the backend and python code seems to be fixated on keytype id 1
+    // senderKeyType,
+    fee: fee(paramsOrTx, 100000000),
+    timestamp: paramsOrTx.timestamp || Date.now(),
+    proofs: paramsOrTx.proofs || [],
+    associationType: paramsOrTx.associationType,
+    hash: paramsOrTx.hash ? paramsOrTx.hash : '',
+    recipient: paramsOrTx.recipient,
+    expires: paramsOrTx.expires || 0,
+    sponsorPublicKey,
+  }
+
+  const tx = version == 3 ? v3Tx : v1Tx
   const bytes = binary.serializeTx(tx)
 
   seedsAndIndexes.forEach(([s, i]) => addProof(tx, signBytes(bytes, s), i))
   tx.id = hashBytes(bytes)
 
+  if (tx === v3Tx && sponsorPublicKey) {
```

Comment:
Is this needed? won't `sponsorPublicKey` already be there because of line 58?

Reply:
Yupz, good catch. Was a remnant of when things failed and I hadn't looked at the actual serialization process.
Changed it as well as for the anchor transaction